tests/CMakeLists.txt

```diff
@@ -122,7 +122,8 @@ set(PYBIND11_TEST_FILES
     test_stl_binders.cpp
     test_tagbased_polymorphic.cpp
     test_union.cpp
-    test_virtual_functions.cpp)
+    test_virtual_functions.cpp
+    test_thread.cpp)
```

Comment:
Does this need to be in it's own file? Can you quickly argue/explain why, and why it cannot fit into one of the existing test files?

Also, please sort alphabetically, here. And should it be `test_threading.cpp`? (that last one really is nitpicking, but it's testing things related to threading, and not really some `thread` class itself?)

Reply:
It probably could be in another file. I'm not certain it's best as a unit test, as it creates threads and pybind11 supports many platforms, versions etc. and threading (and c++) is different across them. I'm happy to take suggestions.

Sorry about the sorting - will fix.

I agree with the filename too.

Reply:
I would recommend it be combined with an existing file, as a new file could increase the build time a bit (unless it really needs to be separate).

Reply:
> Sorry about the sorting - will fix.

No worries. That's why we're reviewing. And if the core is good, we start whining about this, ofc ;-)

The threads seem to be fully cleaned up again, right? So if this can be part of the `test_iostream` files, I think that would be fine, to me :-)

Reply:
ok - how about test_iostream? Seems pretty relevant.

Reply:
Perfect, great idea ;-) Seems our multithreading needs some kind of synchronization as well?

But yes, then the name also makes perfect sense: `test_threading` as part of `test_iostream`